src/cowboy_clock.erl

```diff
@@ -83,10 +83,12 @@ handle_cast(_Msg, State) ->
 	{noreply, State}.
 
 -spec handle_info(any(), State) -> {noreply, State} when State::#state{}.
-handle_info(update, #state{universaltime=Prev, rfc1123=B1, tref=TRef}) ->
+handle_info(update, #state{universaltime=Prev, rfc1123=B1, tref=TRef0}) ->
+	erlang:cancel_timer(TRef0),
```

Comment:
I think that you are wrong.  If I send another message  update to this cowboy_clock, I will create a second wave of timers: there will exist two timers that will send update to this process and they will leave in different waves.

This line is very important because it removes appearing of second wave of timers.